File: sisuAnalise/src/main/java/sisu/Dados.java
package sisu;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.ArrayList;

//padrão Singleton : permite criar objetos únicos para os quais há apenas uma instância
public class Dados {
  
    private static Dados instancia;
    private ArrayList<Candidato> listaCandidatos;
    
    //Construtor que faz a leitura do arquivo
    private Dados() {
        this.listaCandidatos = new ArrayList<>();
        String caminhoArquivo = "sisuDados.csv";
        InputStream is = this.getClass().getClassLoader().getResourceAsStream(caminhoArquivo);
        
        if (is == null) {
            System.err.println("Erro: O arquivo " + caminhoArquivo + " não foi encontrado no classpath.");
            return; 
        }

        try (BufferedReader leitor = new BufferedReader(new InputStreamReader(is))) {
            String linha;
            leitor.readLine();
            while ((linha = leitor.readLine()) != null) {
                String[] dados = linha.split(",");
                Candidato candidato = new Candidato(dados); 
                this.listaCandidatos.add(candidato);
            }
        } catch (IOException e) {
            System.err.println("Erro ao ler o arquivo CSV: " + caminhoArquivo);
            e.printStackTrace(); 
        }
    }
    
    // Método estático para obter a única instância e, portanto, os dados já carregados
    public static Dados getInstancia() {
        if (instancia == null) {
            instancia = new Dados();
        }
        return instancia;
    }
    // Método para obter a lista de candidatos
    public ArrayList<Candidato> getListaCandidatos() {
        return this.listaCandidatos;
    }

}

File: sisuPy/dados.py
import os
from typing import List
from candidato import Candidato
import importlib.resources as pkg_resources

class Dados:
    __instancia = None

    def __init__(self):
        self.__lista_candidatos: List[Candidato] = []
        self._load_csv()

    def _load_csv(self):
        caminho_arquivo = os.path.join("resources", "sisuDados.csv")
        try:
            with open(caminho_arquivo, 'r', encoding='utf-8') as f:
                for linha in f:
                        partes = linha.strip().split(",")
                        candidato = Candidato(partes)
                        self.__lista_candidatos.append(candidato)
        except FileNotFoundError:
            print(f"Erro: O arquivo não foi encontrado.")
            # Você pode levantar uma exceção aqui ou lidar com o erro
        except Exception as e:
            print(f"Ocorreu um erro ao ler o arquivo: {e}")

    @classmethod
    def get_instancia(cls):
        if cls.__instancia is None:
            cls.__instancia = cls()
        return cls.__instancia

    def get_lista_candidatos(self) -> List[Candidato]:
        return self.__lista_candidatos
